tests/integration/pilot/sidecartls/main_test.go

```diff
@@ -0,0 +1,356 @@
+		DNSDomain:   appNamespace.Name() + ".cluster.local",
+		Type:        model.SidecarProxy,
+		IPAddresses: []string{"1.1.1.1"},
+		Metadata: &model.NodeMetadata{
+			Namespace: appNamespace.Name(),
+		},
+		ConfigNamespace: appNamespace.Name(),
+	}
+	return p, nodeID, func() {
+		// allow disable mtls later
+		deleteConfig(t, g, config, StricteMtls, appNamespace)
+		createConfig(t, g, config, DisableMTLS, appNamespace)
+	}
+}
+
+func createConfig(t *testing.T, g galley.Instance, config Config, yaml string, namespace namespace.Instance) {
+	tmpl, err := template.New("Config").Parse(yaml)
+	if err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, config); err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	if err := g.ApplyConfig(namespace, buf.String()); err != nil {
+		t.Fatalf("failed to apply config: %v. Config: %v", err, buf.String())
+	}
+}
+
+func deleteConfig(t *testing.T, g galley.Instance, config Config, yaml string, namespace namespace.Instance) {
+	tmpl, err := template.New("Config").Parse(yaml)
+	if err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, config); err != nil {
+		t.Errorf("failed to create template: %v", err)
+	}
+	if err := g.DeleteConfig(namespace, buf.String()); err != nil {
+		t.Fatalf("failed to delete config: %v. Config: %v", err, buf.String())
+	}
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("sidecar_custom_tls_test", m).
+		RequireEnvironment(environment.Native).
+		Run()
+}
+
+func TestSidecarCustomTLS(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		configFn := func(c Config) Config {
+			return c
+		}
+		p, nodeID, applyConfig := setupTest(t, ctx, configFn)
+		var nodeMetadata = &structpb.Struct{Fields: map[string]*structpb.Value{
+			"ISTIO_VERSION": {Kind: &structpb.Value_StringValue{StringValue: "1.3"}},
+			"NAMESPACE":     {Kind: &structpb.Value_StringValue{StringValue: nodeID.Metadata.Namespace}},
+		}}
+		listenerReq := &xdsapi.DiscoveryRequest{
+			Node: &xdscore.Node{
+				Id:       nodeID.ServiceNode(),
+				Metadata: nodeMetadata,
+			},
+			TypeUrl: v2.ListenerType,
+		}
+
+		if err := p.StartDiscovery(listenerReq); err != nil {
+			t.Fatal(err)
+		}
+		if err := p.WatchDiscovery(5*time.Second, checkInboundFilterChainTLS); err != nil {
+			t.Error(err)
+		}
+		// apply PeerAuthentication to disable mtls
+		applyConfig()
+		if err := p.WatchDiscovery(5*time.Second, checkCustomInboundFilterChainTLS); err != nil {
+			t.Error(err)
+		}
+	})
+}
+
+// Check inbound filter chain tls context
+func checkInboundFilterChainTLS(resp *xdsapi.DiscoveryResponse) (success bool, e error) {
+	expected := map[string]struct{}{
+		"1.1.1.1_9080":    {},
+		"0.0.0.0_80":      {},
+		"virtualInbound":  {},
+		"virtualOutbound": {},
+	}
+
+	expectedTLSContext := &auth.DownstreamTlsContext{
```

Comment:
This seems like an awfully heavy weight way to run this test. If all we are doing is creating configs and then checking XDS response, not running any real pods/traffic, can't we just do a unit test?

Reply:
For this case it is a little painful to run real test for traffic, so i do this as some other cases.

Reply:
https://github.com/istio/istio/blob/master/tests/integration/pilot/sidecarscope/scope_serviceentry_dns_test.go

https://github.com/istio/istio/blob/master/tests/integration/pilot/envoyfilter/main_test.go

Reply:
Yeah it looks verbose but the alternative is to have one-way TLS certs in the sidecar (one that is not Istio cert). That will essentially involve lot more machinery to test this functionality, compared to what is currently in this test IMO.